tmt/cli.py

```diff
@@ -1413,6 +1413,10 @@ def status(
         raise tmt.utils.GeneralError(
             "Options --abandoned, --active and --finished cannot be "
             "used together.")
+    # Re-initialize workdir root in case env TMT_WORKDIR_ROOT was ignored
+    if workdir_root == str(tmt.utils.WORKDIR_ROOT):
+        workdir_root = tmt.utils.get_workdir_root()
+        context.params['workdir_root'] = workdir_root
```

Comment:
I would drop the option completely, TBH. I see no reason for a dedicated option when every other command follows `TMT_WORKDIR_ROOT`. At https://github.com/teemtee/tmt/blob/main/tmt/base.py#L3165, I would replace `self.opt()` call with a call to your new `get_workdir_root()` function.

Reply:
After replacing `self.opt()` call with a call to the new `get_workdir_root()` function. A new issue is seen, e.g.

```bash
(tmt) huanli@kvm-01-guest06:tests$ export TMT_WORKDIR_ROOT=/tmp/foo-tmt-root
(tmt) huanli@kvm-01-guest06:tests$ tmt status -v
status     id
done       /tmp/foo-tmt-root/run-002  /plans/foo
done       /tmp/foo-tmt-root/run-001  /plans/foo
(tmt) huanli@kvm-01-guest06:tests$ unset TMT_WORKDIR_ROOT
(tmt) huanli@kvm-01-guest06:tests$ 
(tmt) huanli@kvm-01-guest06:tests$ tmt status -v
status     id
done       /var/tmp/tmt/run-001  /plans/foo
(tmt) huanli@kvm-01-guest06:tests$ tmt status -v --workdir-root=/tmp/foo-tmt-root
status     id
done       /var/tmp/tmt/run-001  /plans/foo
# ^^^^^^^^^^^^^^^^^^^^^ # should NOT be /var/tmp/tmt
```

`tmt status -v --workdir-root=...` looks bad.

@happz, any suggestion?

Reply:
Yeah, my suggestion would be to remove the option completely :) Your "failing" example is actually correct in this sense, as `TMT_WORKDIR_ROOT` is unset, therefore a default value `/var/tmp/tmt` is used, and `--workdir-root` is ignored because of the added `get_workdir_root()` call I proposed above. Now I would remove the option, which then removes the temptation to run `tmt status --workdir-root...` :)

Reply:
After `--workdir-root` is removed from `tmt status`, the test result looks like:

```bash
(tmt) huanli@kvm-01-guest06:tests$ export TMT_WORKDIR_ROOT=/tmp/foo-tmt-root
(tmt) huanli@kvm-01-guest06:tests$ tmt status -v
status     id
done       /tmp/foo-tmt-root/run-002  /plans/foo
done       /tmp/foo-tmt-root/run-001  /plans/foo
(tmt) huanli@kvm-01-guest06:tests$ unset TMT_WORKDIR_ROOT
(tmt) huanli@kvm-01-guest06:tests$ tmt status -v
status     id
done       /var/tmp/tmt/run-001  /plans/foo
(tmt) huanli@kvm-01-guest06:tests$ tmt status -v --workdir-root=/tmp/foo-tmt-root
Usage: tmt status [OPTIONS]
Try 'tmt status --help' for help.

Error: No such option: --workdir-root
```

And we have some tests to verify `--workdir-root`, e.g.
* https://github.com/teemtee/tmt/blob/main/tests/clean/guests/test.sh#L54

@happz, how should we deal with such a test above?

Reply:
Ok, I didn't know other commands - `clean guests` - also accepts `--workdir-root`. This makes them yet another exception.

@lukaszachy what would be the prefered outcome? Right now, some command have `--workdir-root` and ignore `TMT_ORKDIR_ROOT`, so, shall we:

* drop the option completely, and rely on `TMT_WORKDIR_ROOT` only, or
* promote it to tmt's own option (`tmt clean guests --workdir-root` => `tmt --workdir-root clean guest`) so it becomes accepted by all commands, not just some, and amend its behavior with `TMT_WORKDIR_ROOT`?

Reply:
Hm, workdir-root option makes sense only for 'run', 'status' and 'clean' subcommands, out of which just 'run' doesn't accept it now. 
So I'd say add --workdir-root option to 'run' and make this option respecting TMT_WORKDIR_ROOT

Reply:
Hi @lukaszachy and @happz, what's the priority order of env `TMT_WORKDIR_ROOT` and option `--workdir-root`? From my understanding, the order looks like: `--workdir-root > TMT_WORKDIR_ROOT` . That is, 
1.  Get the value of **option** `--workdir-root` first. If it's empty, goto 2#
2.  Get the value of **env** `TMT_WORKDIR_ROOT`. If it's empty, goto 3#
3.  Get the default value of **global variable** `WORKDIR_ROOT`.

What do you think?

Reply:
Yes, that should be the order of priorities. Could you start simply by adding `env='TMT_WORKDIR_ROOT'` to the `--workdir-root` option? Click is able to extract value from a given environment variable if it has not been specified on command line, which would look like it *was* when inspected in command. That's basically what we'd like to achieve, `--workdir-root` would be set by command line or by an envvar, or not set at all and then the default value of `WORKDIR_ROOT` would apply automagically.

Reply:
Hi @happz, after adding `envvar='TMT_WORKDIR_ROOT'` to the `--workdir-root` option, the test result looks good. e.g.

```bash
(tmt) huanli@KVM01:foo$ export TMT_WORKDIR_ROOT=/tmp/foo-tmt-root
(tmt) huanli@KVM01:foo$ tmt status -v
status     id
done       /tmp/foo-tmt-root/run-002  /plans/foo
done       /tmp/foo-tmt-root/run-001  /plans/foo
(tmt) huanli@KVM01:foo$ tmt status -v --workdir-root=/var/tmp/tmt 
status     id
(tmt) huanli@KVM01:foo$ tmt status -v --workdir-root=/tmp/foo-tmt-root
status     id
done       /tmp/foo-tmt-root/run-002  /plans/foo
done       /tmp/foo-tmt-root/run-001  /plans/foo
(tmt) huanli@KVM01:foo$ 
(tmt) huanli@KVM01:foo$ unset TMT_WORKDIR_ROOT
(tmt) huanli@KVM01:foo$ tmt run 
/var/tmp/tmt/run-001
...<snip>..............................
total: 1 test passed
(tmt) huanli@KVM01:foo$ tmt status -v
status     id
done       /var/tmp/tmt/run-001  /plans/foo
(tmt) huanli@KVM01:foo$ 
(tmt) huanli@KVM01:foo$ tmt status -v --workdir-root=/var/tmp/tmt
status     id
done       /var/tmp/tmt/run-001  /plans/foo
(tmt) huanli@KVM01:foo$ tmt status -v --workdir-root=/tmp/foo-tmt-root
status     id
done       /tmp/foo-tmt-root/run-002  /plans/foo
done       /tmp/foo-tmt-root/run-001  /plans/foo
```

Do we need to add a test case to verify the order of priorities of `tmt status`?

Reply:
Hi @happz, a simple test is added to [tests/status/base/test.sh](https://github.com/teemtee/tmt/blob/main/tests/status/base/test.sh), please help to review, thanks!